CMakeLists.txt

```diff
@@ -65,6 +76,25 @@ execute_process(COMMAND openassetio-traitgen ${PROJECT_BINARY_DIR}/traits.yml
                 COMMAND_ERROR_IS_FATAL ANY
                 COMMAND_ECHO STDERR)
 
+if (OPENASSETIO_MEDIACREATION_GENERATE_PYTHON)
+    # Generate the python package by running traitgen.
+    # At the time of writing, this produces the same artifact as doing
+    # `pip install .` (and invoking setup.py), but that's not a
+    # guarentee going forward, as additional elements of the python
+    # package may be added to mediacreation.
+    #
+    # Furthermore, as this uses the traitgen available in the
+    # environment, rather than a python dependency resolution as pip
+    # would, there is no guarentee that this uses the same traitgen
+    # version a `pip install .` would, so the packages are not
+    # guarenteed to be identical between cmake/pip python package
+    # creation methods.
+    execute_process(COMMAND openassetio-traitgen ${PROJECT_BINARY_DIR}/traits.yml
+    -o ${PROJECT_BINARY_DIR}/package -g python
```

Comment:
[consider] I wonder if we should tidy up the binary (build) dir a bit, i.e. put the python files in a `python` directory and the C++ files in a `cpp` directory, rather than C++ going in the root and Python going in `package`

Reply:
Yeah, it's smart to keep this organised. Was focused on dist, didn't really think about how the build dir would look. https://github.com/OpenAssetIO/OpenAssetIO-MediaCreation/pull/50/commits/fb98fa51ba42a07e833a3f27e58aebf7a9b9b669

Reply:
Much prettier, thanks.